src/base/static/components/templates/map.tsx

```diff
@@ -109,7 +174,7 @@ class MapTemplate extends Component {
     //     be in progress.
     // "error": An error occurred when fetching data for this source.
     mapSourcesLoadStatus: this.props.mapSourceNames.reduce(
-      (memo, groupName) => ({
+      (memo, groupName: any) => ({
```

Comment:
```suggestion
      (memo, groupName: string) => ({
```

You can make that change if you update the prop type above like so:

```js
mapSourceNames: PropTypes.arrayOf(PropTypes.string.isRequired).isRequired,
```
And if the string is nullable, then we would want to do something like so:

```js
      (memo, groupName: string | null) => {
        if (!groupName) {
          groupName = "no-group-name";
        }
        return {
          ...memo,
          [groupName]: "unloaded",
        };
      },

```

Hopefully this answers the question in our DM, and clarifies the types of errors the compiler is protecting us against.

Reply:
I see-- yeah, that makes sense. 👍